src/main/java/frc/robot/subsystems/Drivetrain/DriveImpl.java

```diff
@@ -241,11 +240,11 @@ public void resetEncoderTicks() {
         return Pair.of(this.leftEncoder.getPosition() , this.rightEncoder.getPosition());
     }
 
-    //public void setSetpoint(double setpoint){
-    //    this.leftPIDController.setSetpoint(setpoint);
-    //    this.rightPIDController.setSetpoint(setpoint);
-    //    
-    //}
+    public void setSetpoint(double leftSetPoint, double rightSetPoint){
+       this.leftPIDController.setReference(leftSetPoint, ControlType.kPosition);
+       this.rightPIDController.setReference(rightSetPoint, ControlType.kPosition);
```

Comment:
There is currently separate pid constants defined as class variables in Autonomous.java, I assumed it would have been more simple to just tune seperate constants. I believe we could also use a conversion from displacement to velocity  through an average acceleration calculated by PID but my math could be way off on that. Regardless the inner and outer loop sounds like a good idea!

Reply:
You're going to need separate constants in either case, but the difference is whether there is one loop or two. You could compute ticks to velocity on your own and roll your own controller that tries to do both position and velocity closed loop control at once, or you can just use the already developed and well-known PID controllers and simply feed the output of one into the input of the other.